### PR TITLE
Update CoreFX dependencies

### PR DESCRIPTION
**Pull request by tmat**
StreamJsonRpc requires the higher versions of these packages.

Removes hard-coded list of suggested binding redirects and replaces it with a generated list.

### CODE REVIEW

**Comment by tmat** [comment]
@heejaechang @AArnott

**Comment by tmat** [comment]
@jaredpar PTAL
@JoeRobich FYI

**Comment by tmat** [comment]
@nguerrera PTAL

File: eng/targets/GenerateCompilerExecutableBindingRedirects.targets [hunk 1]
```diff
@@ -1,56 +1,20 @@
...
-      <SuggestedBindingRedirects Include="System.Runtime.CompilerServices.Unsafe, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-        <MaxVersion>4.0.4.1</MaxVersion>
-      </SuggestedBindingRedirects>
-      <SuggestedBindingRedirects Include="System.Text.Encoding.CodePages, Version=0.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a">
-        <MaxVersion>4.1.1.0</MaxVersion>
-      </SuggestedBindingRedirects>
-      <SuggestedBindingRedirects Include="System.Threading.Tasks.Extensions, Version=0.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51">
-        <MaxVersion>4.2.0.0</MaxVersion>
-      </SuggestedBindingRedirects>
+      <SuggestedBindingRedirects Include="@(ReferencePath->'%(FusionName)')" MaxVersion="%(ReferencePath.Version)" KeepMetadata="-none-" Condition="'%(ReferencePath.FrameworkFile)' != 'true'" />
```
Comment (nguerrera) [inline eng/targets/GenerateCompilerExecutableBindingRedirects.targets]: Can you share a diff of config file before and after this change?

Comment (tmat) [reply eng/targets/GenerateCompilerExecutableBindingRedirects.targets]: It's the same as the one in previous commit:
https://github.com/dotnet/roslyn/pull/33993/commits/6e54564f7e4c5e6404f2a42eedf9b50afbb79586

Comment (tmat) [reply eng/targets/GenerateCompilerExecutableBindingRedirects.targets]: I mean the values are.

**Pull request merged**